fabric/dataclient.go

```diff
@@ -495,170 +413,41 @@ func convertOne(fg *Fabric) ([]*eskip.Route, error) {
-					Path:   p.Path,
-					Method: strings.ToUpper(m.Method),
-					Predicates: []*eskip.Predicate{
-						{
-							Name: predicates.HostAnyName,
-							Args: []interface{}{
-								host,
-								host + ":443",
-							},
-						},
-						{
-							Name: predicates.HeaderName,
-							Args: []interface{}{
-								"X-Forwarded-Proto",
-								"https",
-							},
-						},
-						{
-							Name: predicates.WeightName,
-							Args: []interface{}{
-								23, // TODO(sszuecs) needs checking
-							},
-						},
-					},
-					Filters: []*eskip.Filter{
-						{
-							// oauthTokeninfoAnyKV("realm", "/services", "realm", "/employees")
-							Name: filters.OAuthTokeninfoAnyKVName,
-							Args: []interface{}{
-								// TODO(sszuecs): should be configurable
-								"realm",
-								"/services",
-								"realm",
-								"/employees",
-							},
-						},
-						{
-							// oauthTokeninfoAllScope("uid", "spp-brand-service.brands.write")
-							Name: filters.OAuthTokeninfoAllScopeName,
-							Args: privs,
-						},
-						{
-							// unverifiedAuditLog("sub")
-							Name: filters.UnverifiedAuditLogName,
-							Args: []interface{}{
-								// TODO(sszuecs): in the future should be configurable
-								"sub",
-							},
-						},
-					},
-					BackendType: eskipBackend.Type,
-					Backend:     eskipBackend.backend,
-					LBAlgorithm: eskipBackend.lbAlgorithm,
-					LBEndpoints: eskipBackend.lbEndpoints,
-				}
-
-				// add optional ratelimit
-				if m.Ratelimit != nil {
-					r.Filters = append(r.Filters,
-						&eskip.Filter{
-							//inlineContentIfStatus(429, "{\"title\": \"Rate limit exceeded\", \"detail\": \"See the retry-after header for how many seconds to wait before retrying.\", \"status\": 429}", "application/problem+json")
-							Name: filters.InlineContentIfStatusName,
-							Args: []interface{}{
-								429,
-								"{\"title\":\"Rate limit exceeded\",\"detail\":\"See the retry-after header for how many seconds to wait before retrying.\",\"status\":429}",
-								"application/problem+json",
-							},
-						},
-						&eskip.Filter{
-							// clusterClientRatelimit("spp-brand-service_api-brand-assignments-id_DELETE", 30, "1m", "Authorization")
-							Name: filters.ClusterClientRatelimitName,
-							Args: []interface{}{
-								fmt.Sprintf("%s_%s_%s",
-									fg.Metadata.Name,
-									strings.Trim(nonWord.ReplaceAllString(p.Path, "-"), "-"),
-									m.Method,
-								),
-								m.Ratelimit.DefaultRate,
-								m.Ratelimit.Period,
-								// optional header, TODO(sszuecs): maybe configurable in the future
-								"Authorization",
-							},
-						},
-					)
+				// local allowlist overrides default. In case allow list is disabled only use scopes.
+				//     example: oauthTokeninfoAnyKV("sub", "app1", "sub", "app2", ..)
+				var allowedServices []interface{}
+				disableAllowList := false
+				if m.AllowList != nil {
+					switch m.AllowList.State {
```

Comment:
This `switch` can be converted to a regular `if` right? Because basically there is only a case, where `state == "disabled"` or the default. Namely, `if m.AllowList.State == "disabled" { disableAllowList = true } else { .. }`